extlib/millhone/src/cmd/commit.rs

```diff
@@ -0,0 +1,255 @@
+use std::{collections::HashSet, path::PathBuf};
+
+use clap::{Parser, ValueEnum};
+use getset::Getters;
+use itertools::Itertools;
+use millhone::extract::{Kind, Method, Target, Transform};
+use serde::{Deserialize, Serialize};
+use srclib::{Fetcher, Locator};
+use stable_eyre::{
+    eyre::{bail, eyre, Context},
+    Report,
+};
+use strum::{Display, IntoEnumIterator};
+use tap::{Pipe, Tap, TapFallible};
+use tracing::{debug, info, warn};
+
+use crate::cmd::MatchingSnippet;
+
+/// Options for snippet ingestion.
+#[derive(Debug, Parser, Getters)]
+#[getset(get = "pub")]
+#[clap(version)]
+pub struct Subcommand {
+    /// The path to the directory in which the `analyze` subcommand wrote its output.
+    #[clap(long)]
+    analyze_output_dir: PathBuf,
+
+    /// The output format for the generated `fossa-deps` file.
+    #[clap(long, default_value_t = OutputFormat::Json)]
+    format: OutputFormat,
+
+    /// If specified, overwrites the output file if it exists.
+    #[clap(long)]
+    overwrite_fossa_deps: bool,
+
+    /// Only commit matches with the specified targets.
+    ///
+    /// Defaults to all supported targets.
+    /// Specify the argument multiple times to indicate additional options.
+    #[clap(long = "target")]
+    targets: Vec<Target>,
+
+    /// Only commit matches with the specified kinds.
+    ///
+    /// Defaults to all supported kinds.
+    /// Specify the argument multiple times to indicate additional options.
+    #[clap(long = "kind")]
+    kinds: Vec<Kind>,
+
+    /// Only commit matches with the specified transforms.
+    ///
+    /// Defaults to all supported transforms. Raw matches (i.e. no transforms)
+    /// are always committed and cannot be disabled.
+    /// Specify the argument multiple times to indicate additional options.
+    #[clap(long = "transform")]
+    transforms: Vec<Transform>,
+
+    /// The directory being analyzed by FOSSA.
+    ///
+    /// The `fossa-deps` file is written into the root of this directory.
+    target: PathBuf,
+}
+
+/// The output format for the generated `fossa-deps` file.
+#[derive(Debug, Clone, Copy, ValueEnum, Display)]
+#[strum(serialize_all = "snake_case")]
+pub enum OutputFormat {
+    /// Commit as `fossa-deps.json`.
+    Json,
+
+    /// Commit as `fossa-deps.yml`.
+    Yml,
+}
+
+#[tracing::instrument(skip_all, fields(target = %opts.target().display()))]
+pub fn main(opts: Subcommand) -> Result<(), Report> {
+    info!(
+        analyze_output_dir = ?opts.analyze_output_dir,
+        format = %opts.format,
+        targets = ?opts.targets,
+        kinds = ?opts.kinds,
+        transforms = ?opts.transforms,
+        "Committing local snippet matches",
+    );
+
+    let output_file = opts
+        .target()
+        .join("fossa-deps")
+        .with_extension(opts.format().to_string());
+    if !opts.overwrite_fossa_deps() && output_file.exists() {
```

Comment:
[Optional] Accepting `-` on the command line to explicitly ask for output be written to stdout may make this a bit friendlier for scripting. Right now for an equivalent I think they'd need some combination of generating unique directories and/or using `--overwrite-output`  and then reading the files.

Reply:
Good point! https://fossa.atlassian.net/browse/ANE-1203